Sources/Commands/SwiftTestTool.swift

```diff
@@ -690,14 +713,34 @@ final class TestRunner {
     }
 }
 
+extension DispatchTimeInterval {
+    /// Encode the nanoseconds dispatch time in JSON
+    /// - Returns: JSON instance for the nanoseconds or `.null`
+    public func toNanosecondsJSON() -> JSON {
+        guard let nanoseconds = nanoseconds() else {
+            return .null
+        }
+        return nanoseconds.toJSON()
+    }
+}
+
 /// A class to run tests in parallel.
 final class ParallelTestRunner {
     /// An enum representing result of a unit test execution.
-    struct TestResult {
+    struct TestResult: JSONSerializable {
```

Comment:
ideally all the JSONSerializable conformances (or better, Encodable) is done in an extension

Reply:
+1 to this, I'm also not sure why `Encodable` is not used here instead of adding a new dependency on TSC

Reply:
Sure thing, is now updated to use `Encodable`